src/web/backend/nextcloud/utils.py

```diff
@@ -50,9 +50,7 @@ def get_group_folder(mount_point):
     Returns:
     """
     nxc = get_nextcloud()
-    group_folders = nxc.get_group_folders().data
+    group_folders = nxc.get_group_folders(mount_point=mount_point).data
```

Comment:
For starters, please keep the mount point filtering in TEAP a separate function in - you can port it to Nextcloud API later.

Reply:
reverted it to filter in iteration as it was before